src/c++/perf_analyzer/triton_metrics_manager.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "triton_metrics_manager.h"
+#include <stdexcept>
+
+namespace triton { namespace perfanalyzer {
+
+TritonMetricsManager::TritonMetricsManager(
+    std::shared_ptr<clientbackend::ClientBackend> client_backend,
+    uint64_t triton_metrics_interval_ms)
+    : client_backend_(client_backend),
+      triton_metrics_interval_ms_(triton_metrics_interval_ms)
+{
+}
+
+TritonMetricsManager::~TritonMetricsManager()
+{
+  StopQueryingTritonMetrics();
+}
+
+void
+TritonMetricsManager::StartQueryingTritonMetrics()
+{
+  should_keep_querying_ = true;
+  query_loop_future_ = std::async(
+      &TritonMetricsManager::QueryTritonMetricsEveryNMilliseconds, this);
+}
+
+void
+TritonMetricsManager::QueryTritonMetricsEveryNMilliseconds()
+{
+  while (should_keep_querying_) {
+    const auto& start{std::chrono::system_clock::now()};
+
+    TritonMetrics triton_metrics{};
+    clientbackend::Error err{client_backend_->TritonMetrics(triton_metrics)};
+    if (err.IsOk() == false) {
+      throw std::runtime_error(err.Message());
+    }
+    triton_metrics_per_timestamp_.emplace_back(
```

Comment:
It looks like there is no way to access these metrics from the public methods?

Reply:
Correct, a future PR will allow access to the data 👍